content/en/docs/reference/kubectl/_index.md

```diff
@@ -1,5 +1,551 @@
+```
+
+## Examples: Creating and using plugins
+
+Use the following set of examples to help you familiarize yourself with writing and using `kubectl` plugins:
+
+```shell
+# create a simple plugin in any language and name the resulting executable file
+# so that it begins with the prefix "kubectl-"
+cat ./kubectl-hello
+```
+```shell
+#!/bin/sh
+
+# this plugin prints the words "hello world"
+echo "hello world"
+```
+With a plugin written, let's make it executable:
+```bash
+chmod a+x ./kubectl-hello
+
+# and move it to a location in our PATH
+sudo mv ./kubectl-hello /usr/local/bin
+sudo chown root:root /usr/local/bin
+
+# You have now created and "installed" a kubectl plugin.
+# You can begin using this plugin by invoking it from kubectl as if it were a regular command
+kubectl hello
+```
+```
+hello world
+```
+
+```shell
+# You can "uninstall" a plugin, by removing it from the folder in your
+# $PATH where you placed it
+sudo rm /usr/local/bin/kubectl-hello
+```
+
+In order to view all of the plugins that are available to `kubectl`, use
+the `kubectl plugin list` subcommand:
+
+```shell
+kubectl plugin list
+```
+The output is similar to:
+```
+The following kubectl-compatible plugins are available:
+
+/usr/local/bin/kubectl-hello
+/usr/local/bin/kubectl-foo
+/usr/local/bin/kubectl-bar
+```
+
+`kubectl plugin list` also warns you about plugins that are not
+executable, or that are shadowed by other plugins; for example:
+```shell
+sudo chmod -x /usr/local/bin/kubectl-foo # remove execute permission
+kubectl plugin list
+```
+```
+The following kubectl-compatible plugins are available:
+
+/usr/local/bin/kubectl-hello
+/usr/local/bin/kubectl-foo
+  - warning: /usr/local/bin/kubectl-foo identified as a plugin, but it is not executable
+/usr/local/bin/kubectl-bar
+
+error: one plugin warning was found
+```
+
+You can think of plugins as a means to build more complex functionality on top
+of the existing kubectl commands:
+
+```shell
+cat ./kubectl-whoami
+```
+The next few examples assume that you already made `kubectl-whoami` have
+the following contents:
+```shell
+#!/bin/bash
+
+# this plugin makes use of the `kubectl config` command in order to output
+# information about the current user, based on the currently selected context
+kubectl config view --template='{{ range .contexts }}{{ if eq .name "'$(kubectl config current-context)'" }}Current user: {{ printf "%s\n" .context.user }}{{ end }}{{ end }}'
+```
+
+Running the above command gives you an output containing the user for the
+current context in your KUBECONFIG file:
+
+```shell
+# make the file executable
+sudo chmod +x ./kubectl-whoami
+
+# and move it into your PATH
+sudo mv ./kubectl-whoami /usr/local/bin
+
+kubectl whoami
+Current user: plugins-user
+```
+
+## {{% heading "whatsnext" %}}
+
+* Read the `kubectl` [command reference](/docs/reference/kubectl/kubectl/).
+* Read the `kubectl` [command line arguments](/docs/reference/kubectl/kubectl/) reference.
```

Comment:
👋 @sftim, Can you check the links for the first two items (links are the same)?
The page could be configured to ignore the list of section links at the bottom of the page, or do you think the links add value?

Reply:
Good spot.